### DISCUSSION TITLE
RuleTester use-case with multiple files

### DISCUSSION THREAD

**Discussion by alexnix** [eslint/eslint#15388] [Q&A]
I have a very specific use-case for which i wrote a rule to test it. The use-case is: if a variable of type `number`, with a fixed name (say `foo`) is imported from a module with a fixed name (say `bar`), then I am reporting this in the rule.

So the code following code would be reported
```
import {foo} from 'bar';  
```

if in module `bar`:
```
export const foo: number = 120;
```

I was able to implement this using the type checker and the `@typescript-eslint/parser` parser.

BUT, I am having trouble testing this. 
If I create the two files, and read the content of the one with the import, then pass it the code property of a test, it will not be able to tell the type of the variable (will see it as `any`). I assume it does not do the full parsing/compilation like it does when actually running the rule.

Any thoughts on how to test this kind of rule?

PS: the `bar` module may not be lint-ed itself, as in the real use-case it is not an actual file in the same project, but rather a separate module in node_modules. And the linter does not go that deep as to lint files from the node_modules (and I don't want it it to); tho apparently while processing the rule in a real case, it knows the type correctly.

PS: when running the RulteTester, it does not get the type, even if `bar` is a file in the same project.

**Comment by nzakas** [comment]
Rules aren’t meant to work across multiple files, and we don’t work on the typescript integration, so I’m afraid we can’t offer any advice on how typing works. You’d be better off asking the typescript-eslint folks.

**Comment by alanperezg** [comment]
You can use [mock-fs](https://www.npmjs.com/package/mock-fs) to mock the file system on your tests.
```ts
describe("my test suite", () => {
  beforeEach(() => {
    mockfs({
      "bar.js":  "export const foo: number = 120;",
      "bar2.js": "export const foo: number = 120;"
    });
  })

  afterEach(() => {
    mockfs.restore(); 
  });

  ruleTester.run("eslint rule tester", myRule, {
    valid: [{
      code: "import {foo} from 'bar2';",
    }],
    invalid: [{
      code: "import {foo} from 'bar';",
      errors: [{ messageId: "<your-error-msg>" }]
    }],
  });
});
```
